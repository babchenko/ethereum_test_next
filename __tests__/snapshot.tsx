import { render } from '@testing-library/react'
import Home from 'pages/index'
import { mockCitizens } from './mock/mockData';
import { createMockRouter } from './mock/createRouter';
import { RouterContext } from 'next/dist/shared/lib/router-context';

it('renders homepage unchanged', () => {
  const { container } = render(
      <RouterContext.Provider value={createMockRouter({ query: { page: '1' } })}>
        <Home citizens={mockCitizens} total={mockCitizens.length} page={1} />
      </RouterContext.Provider>
  )
  expect(container).toMatchSnapshot()
})
