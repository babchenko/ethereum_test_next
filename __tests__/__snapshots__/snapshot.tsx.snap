// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <div
      class="ui container"
    >
      <div
        class="ui menu"
        style="margin-top: 10px;"
      >
        <a
          class="item"
          href="/"
        >
          FETestTask
        </a>
        <div
          class="right menu"
        >
          <a
            class="item"
            href="/new"
          >
            +
          </a>
        </div>
      </div>
      <table
        class="ui celled table"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class=""
            >
              ID
            </th>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Age
            </th>
            <th
              class=""
            >
              City
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
            style="cursor: pointer;"
          >
            <td
              class=""
              data-testid="id"
            >
              fake-id-1
            </td>
            <td
              class=""
              data-testid="name"
            >
              fake-name-1
            </td>
            <td
              class=""
              data-testid="age"
            >
              fake-age-1
            </td>
            <td
              class=""
              data-testid="city"
            >
              fake-city-1
            </td>
          </tr>
          <tr
            class=""
            style="cursor: pointer;"
          >
            <td
              class=""
              data-testid="id"
            >
              fake-id-2
            </td>
            <td
              class=""
              data-testid="name"
            >
              fake-name-2
            </td>
            <td
              class=""
              data-testid="age"
            >
              fake-age-2
            </td>
            <td
              class=""
              data-testid="city"
            >
              fake-city-2
            </td>
          </tr>
        </tbody>
        <tfoot
          class=""
        >
          <tr
            class=""
          >
            <th
              class=""
              colspan="4"
            >
              <div
                aria-label="Pagination Navigation"
                class="ui pagination menu"
                role="navigation"
              >
                <a
                  aria-current="false"
                  aria-disabled="false"
                  aria-label="First item"
                  class="item"
                  tabindex="0"
                  type="firstItem"
                  value="1"
                >
                  «
                </a>
                <a
                  aria-current="false"
                  aria-disabled="false"
                  aria-label="Previous item"
                  class="item"
                  tabindex="0"
                  type="prevItem"
                  value="1"
                >
                  ⟨
                </a>
                <a
                  aria-current="false"
                  aria-disabled="false"
                  aria-label="Next item"
                  class="item"
                  tabindex="0"
                  type="nextItem"
                  value="0"
                >
                  ⟩
                </a>
                <a
                  aria-current="false"
                  aria-disabled="false"
                  aria-label="Last item"
                  class="item"
                  tabindex="0"
                  type="lastItem"
                  value="0"
                >
                  »
                </a>
              </div>
            </th>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
